lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -141,6 +141,16 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             # therefore, this command will always exit 0 if either service is installed
             on host, Command.new("sc query puppet || sc query pe-puppet", [], { :cmdexe => true })
 
+            # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
+            # registry and honored after install/upgrade.
+            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
+            on host, Command.new(reg_query_command, [], { :cmdexe => true })
+
+            start_mode = msi_opts['PUPPET_AGENT_STARTUP_MODE'] == "Automatic" ? "Auto" : msi_opts['PUPPET_AGENT_STARTUP_MODE']
+            service_query_command = %Q('WMIC SERVICE where (Name like "%Puppet" AND StartMode="#{start_mode}") | findstr Puppet')
```

Comment:
@Iristyle just to confirm - is our evaluation of the user supplied value for `PUPPET_AGENT_STARTUP_MODE` case-insensitive? i.e. user can supply Disabled or DISABLED ?